src/io/windows.zig

```diff
@@ -464,15 +462,27 @@ pub const IO = struct {
                         ) callconv(os.windows.WINAPI) os.windows.BOOL;
 
                         // Find the ConnectEx function by dynamically looking it up on the socket.
-                        const connect_ex = os.windows.loadWinsockExtensionFunction(
-                            LPFN_CONNECTEX,
+                        var connect_ex: LPFN_CONNECTEX = undefined;
+                        var num_bytes: os.windows.DWORD = undefined;
+                        const guid = os.windows.ws2_32.WSAID_CONNECTEX;
+                        switch (os.windows.ws2_32.WSAIoctl(
                             op.socket,
-                            os.windows.ws2_32.WSAID_CONNECTEX,
-                        ) catch |err| switch (err) {
-                            error.OperationNotSupported => unreachable,
-                            error.ShortRead => unreachable,
-                            else => |e| return e,
-                        };
+                            os.windows.ws2_32.SIO_GET_EXTENSION_FUNCTION_POINTER,
+                            @ptrCast(*const anyopaque, &guid),
+                            @sizeOf(os.windows.GUID),
+                            @ptrCast(*anyopaque, &connect_ex),
+                            @sizeOf(LPFN_CONNECTEX),
+                            &num_bytes,
+                            null,
+                            null,
+                        )) {
+                            os.windows.ws2_32.SOCKET_ERROR => switch (os.windows.ws2_32.WSAGetLastError()) {
+                                .WSAEOPNOTSUPP => unreachable,
+                                .WSAENOTSOCK => unreachable,
+                                else => |err| return os.windows.unexpectedWSAError(err),
+                            },
+                            else => assert(num_bytes == @sizeOf(LPFN_CONNECTEX)),
+                        }
```

Comment:
The `T` it takes in needs to be `fn(...)` as it does `&function`, but it stores the variable as `var function: T` (instead of correctly as `var function: *const T`) which errors out on args incorrectly needing comptime. I just reimplemented it inline for now.

Reply:
Ah, so that's a bug in stdlib. Let's add a todo to swith to  the upstream impl once that is fixed.